src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -241,6 +240,17 @@ public static NoThrowTaskAwaitable NoThrowAwaitable(this Task task, bool capture
         return new NoThrowTaskAwaitable(task, captureContext);
     }
 
+    /// <summary>
+    /// Returns an awaitable for the specified <see cref="JoinableTaskFactory.SwitchToMainThreadAsync(CancellationToken)"/>
+    /// operation that will return a Boolean value indicating cancellation rather than throwing an exception.
```

Comment:
📝 The branch I found locally returned `bool` here. If it's acceptable to return `void`, it is likely possible to simplify this pull request.